test/test-lex.cpp

```diff
@@ -118,12 +118,109 @@ TEST(test_lex, lex_binary_numbers) {
+  }
+}
+
+TEST(test_lex, lex_octal_numbers_strict) {
+  check_single_token(u8"000", token_type::number);
+  check_single_token(u8"001", token_type::number);
+  check_single_token(u8"00010101010101010", token_type::number);
+  check_single_token(u8"00010101010101010", token_type::number);
+  check_single_token(u8"051", token_type::number);
+  check_single_token(u8"0o51", token_type::number);
+}
+
+TEST(test_lex, lex_octal_numbers_lax) {
+  check_single_token(u8"058", token_type::number);
+  check_single_token(u8"058.9", token_type::number);
+}
+
+// TODO(👮🏾‍♀️)doesn't work
+TEST(test_lex, fail_lex_octal_numbers_lax) {
+  // Each scope is essentially `check-single-token`, but pulled out of the
+  // function to access the error_collector at the end
+  //
+  // idea: have `check_single_token` accept a callback with the errors
+  {
+    error_collector v;
+    padded_string input(u8"0o58");
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    // q(🎅🏾) have another error kind for
+    // `unexpected_character_in_octal_number?
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unexpected_characters_in_octal_number,
+                              characters, offsets_matcher(&input, 3, 4))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"0o58.2");
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    ASSERT_EQ(v.errors.size(), 1);
+    // q(🎅🏾) have another error kind for
+    // `unexpected_character_in_octal_number?
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unexpected_characters_in_octal_number,
+                              characters, offsets_matcher(&input, 3, 4))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"052.2");
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    // q(🎅🏾) have another error kind for
+    // `unexpected_character_in_octal_number?
+    // edit: did it
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unexpected_characters_in_octal_number,
+                              characters, offsets_matcher(&input, 3, 4))));
+  }
+}
+
+// TODO (👮🏾‍♀️) (when strict mode implemented) tests to fail in
+// strict mode
+
 TEST(test_lex, lex_hex_numbers) {
   check_single_token(u8"0x0", token_type::number);
   check_single_token(u8"0x123456789abcdef", token_type::number);
   check_single_token(u8"0X123456789ABCDEF", token_type::number);
 }
 
+TEST(test_lex, DISABLED_fail_lex_hex_numbers) {
```

Comment:
@sTRAGER request for comment

Reply:
Let's remove this test from this commit. Instead, add a comment to issue #43.

Reply:
[done](https://github.com/quick-lint/quick-lint-js/issues/43#issuecomment-707295300)